src/core/interaction/pickingmapper.cpp

```diff
@@ -89,21 +87,40 @@ void PickingMapper::resize(size_t newSize) {
 }
 
 bool PickingMapper::isEnabled() const {
-    if (pickingAction_)
+    if (pickingAction_) {
         return pickingAction_->isEnabled();
-    else
+    } else {
         return false;
+    }
 }
 
 void PickingMapper::setEnabled(bool enabled) {
     if (pickingAction_) pickingAction_->setEnabled(enabled);
 }
 
-size_t PickingMapper::getPickingId(size_t id) const { return pickingAction_->getPickingId(id); }
+size_t PickingMapper::getPickingId(size_t id) const {
+    if (pickingAction_) {
+        return pickingAction_->getPickingId(id);
+    } else {
+        return 0;
```

Comment:
This might result in errors later because picking id 0 is valid?
Would it be better to throw or assert?
Documentation should be updated :)

Reply:
Hm, valid point. Going for throwing an exception here would be best then.

Reply:
In principle the only time pickingAction_ can be nullptr now is in a moved from object right, then you are not really allowed to call those function anyway...

Reply:
same for the color then

Reply:
So, how should we handle those cases?

Reply:
it's a precondition that's broken. There are a few valid option, do nothing, assert, throw, return something anyway. I don't think we have a standard way of handling this in our guidelines, But I think I would lean towards an assert